From 94f7a071c82f79b66c28445a8def17b1f3ccbcb9 Mon Sep 17 00:00:00 2001
From: kkrishnakumar <kkrishnakumar@maxlinear.com>
Date: Thu, 12 Sep 2024 11:19:07 +0530
Subject: [PATCH] WLANRTSYS-84369 Fix to handle invalid bss_idx updation during
 reconf_add_bss resulting in wrong vapid during mld_setup_link

Issue : Facing mac2index assert and call trace while trying to connect STA for the first time after MLD formation.

Reason : Requesting SID to the same vapID twice resulting in error.

Fix : Removed the updation of bss_idx in reconf_add_bss.

Regression commit : 43af9cda8dd

Unit tested : Yes

FIXUP to WLANRTSYS-60843: MLO changes: AP MLD configuration
---
 src/ap/hostapd.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index a8fe26483..c34272e73 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -4437,7 +4437,6 @@ static int hostapd_reconf_add_bss(struct hostapd_iface *iface, struct hostapd_co
 	}
 
 	idx = iface->num_bss - 1;
-	iface->bss[idx]->conf->bss_idx = idx;
 	hostapd_flush_old_stations(iface->bss[idx], WLAN_REASON_PREV_AUTH_NOT_VALID);
 #ifdef CONFIG_IEEE80211BE_MXL_MLO
 	if(mxl_mld_setup_interfaces(iface->interfaces))
-- 
2.43.0

